plugins/org.python.pydev.ast/src/org/python/pydev/ast/interpreter_managers/InterpreterInfo.java

```diff
@@ -1621,6 +1624,7 @@ public void setEnvVariables(String[] env) {
         if (!hasEnvVarsToUpdate && !this.activateCondaEnv) {
             return env; //nothing to change
         }
+        Map<String, String> condaEnv = new HashMap<String, String>(this.condaEnv);
```

Comment:
Please rename the `this.condaEnv` to `this.condaEnvCache` (so that it's clear that one is the cached version and the other is the local variable version to avoid name shadowing)

Also, please put this inside of `this.activateCondaEnv` as you don't need to access it unless you'll really need to use it (access to `volatile` is slower as it may need to synchronize threads, so, it should be used as little as possible).